src/main/cpp/controllers/DrivetrainController.cpp

```diff
@@ -141,6 +157,21 @@ Eigen::Vector<double, 2> DrivetrainController::Calculate(
     return m_u;
 }
 
+Eigen::Vector<double, 2> DrivetrainController::CalculateTurnInPlace(
+    const Eigen::Vector<double, 3>& x) {
+    m_u = Eigen::Vector<double, 2>::Zero();
+    Eigen::Vector<double, 3> m_turningNextR = {m_nextR[State::kHeading],
+                                               m_nextR[State::kLeftVelocity],
+                                               m_nextR[State::kRightVelocity]};
+    m_u = m_turningLQR.Calculate(x) + m_turningFF.Calculate(m_turningNextR);
```

Comment:
You need a motion profile for the feedforward to help at all. Look at the turret code for an example of using a trapezoid profile.